lib/web_ui/lib/src/engine/util.dart

```diff
@@ -461,11 +461,23 @@ void applyWebkitClipFix(html.Element containerElement) {
 
 final ByteData _fontChangeMessage = JSONMessageCodec().encodeMessage(<String, dynamic>{'type': 'fontsChange'});
 
+// Font load callbacks will typically arrive in sequence, we want to prevent
+// sendFontChangeMessage of causing multiple synchronous rebuilds.
+// This flag ensures we properly schedule a single call to framework.
+bool _fontChangeScheduled = false;
+
 FutureOr<void> sendFontChangeMessage() async {
   if (window._onPlatformMessage != null)
-    window.invokeOnPlatformMessage(
-      'flutter/system',
-      _fontChangeMessage,
-      (_) {},
-    );
+    if (!_fontChangeScheduled) {
+      _fontChangeScheduled = true;
+      // Batch updates into next animationframe.
+      html.window.requestAnimationFrame((num _) {
```

Comment:
This will debounce the notification at 16ms granularity, which seems too low for something that's loading over the network. Maybe use a `Timer` that debounces on the order of 100s of ms? We also have the `AlarmClock` with (snooze) feature for this sort of thing.

Reply:
Gallery loads 6 fonts, they are coming in within 1ms of each other. 1 font sometimes misses 16ms window comes in second. 100ms would delay app first build for many frames, there is no easy signal to cancel the wait and complete future.

Reply:
Gotcha.